src/FuncSharp.Examples/ITry.cs

```diff
@@ -0,0 +1,199 @@
+            ITry<int, NetworkOperationError> flooredMultiplicationResult = multiplicationResult.Map(r => (int)r);
+            ITry<string, NetworkOperationError> stringifiedDivisionResult = multiplicationResult.Map(r => r.ToString());
+        }
+
+        private void TransformingErrorValuesWithMapError(decimal number, decimal multiplier)
+        {
+            ITry<decimal, NetworkOperationError> multiplicationResult = PerformNetworkOperation(_ => number * multiplier);
+
+            // You can use MapError to map for example from exception to another type you use to represent errors. Or map to logging messages.
+            ITry<decimal, string> flooredMultiplicationResult = multiplicationResult.MapError(e => e.ToString());
+        }
+
+        private void HandlingNestedTriesWithFlatMap(decimal number, decimal firstMultiplier, decimal secondMultiplier)
+        {
+            ITry<decimal, NetworkOperationError> multiplicationResult = PerformNetworkOperation(_ => number * firstMultiplier);
+            ITry<ITry<decimal, NetworkOperationError>, NetworkOperationError> resultOfDoubleMultiplication = multiplicationResult.Map(r => PerformNetworkOperation(_ => r * secondMultiplier));
+
+            // This try succeeds only if both tries succeed. However the second lambda is only executed in case the first try is successful.
+            ITry<decimal, NetworkOperationError> flattenedResultOfDoubleDivision = multiplicationResult.FlatMap(r => PerformNetworkOperation(_ => r * secondMultiplier));
+        }
+
+        private void AggregatingMultipleTriesIntoSingleResult(decimal number, decimal multiplier, decimal numberToAdd, decimal numberToSubtract)
+        {
+            // You need to start with tries that have a collection as the type of error. So we MapError to a collection of strings here. But it can be a collection of any type.
+            // For validations, you would write your validation method, so it already returns IEnumerable as the error type.
+            // Try.Aggregate method always executes all tries and then aggregates the success results or the errors. It doesn't stop on first error.
+            // You can transform the result into any class you want. This makes it nice and effective way of parsing values and combining the results.
+            ITry<ClassWithMultiplePropertiesToValidate, IEnumerable<string>> combinedResult = Try.Aggregate(
+                t1: PerformNetworkOperation(_ => number * multiplier).MapError(e => new List<string>{ e.ToString() }),
+                t2: PerformNetworkOperation(_ => number + numberToAdd).MapError(e => new List<string>{ e.ToString() }),
+                t3: PerformNetworkOperation(_ => number - numberToSubtract).MapError(e => new List<string>{ e.ToString() }),
+                f: (multiplicationResult, additionResult, subtractionResult) => new ClassWithMultiplePropertiesToValidate(multiplicationResult, additionResult, subtractionResult)
+            );
```

Comment:
Considering the network operation example doesn't do much sense here, what about some validation example?

Also, why is the result `ToValidate`? Shouldn't it already be valid?